Task6/posts.js

```diff
@@ -175,113 +175,86 @@ var posts = [
         likes: ['Lola Skin', 'Paula', 'Egor B', 'Dean W', 'Sam W']
     }
 ];
-maxId = 20;
-
-function isInteger(num) {
-    return (num ^ 0) === num;
-}
-
-function sortByDateDescending(posts) {
-    posts.sort((p1, p2) => p1.createdAt < p2.createdAt ? 1 : -1);
-}
-
-function sortByDateAscending(posts) {
-    posts.sort((p1, p2) => p1.createdAt > p2.createdAt ? 1 : -1);
-}
 
 (function () {
+    maxId = 20;
+    function sortByDateDescending(posts) {
+        posts.sort((p1,p2) => p2.createdAt-p1.createdAt);
+    }
+
+    function sortByDateAscending(posts) {
+        posts.sort((p1,p2) => p1.createdAt-p2.createdAt);
+    }
+    
+    function filterByAuthors(posts,author) {
+        return posts.filter(function(post) {
+            return post.author===author;
+        })
+    }
+    function filterByHashTags(posts,hashTags) {
+        return posts.filter(function(post) {
```

Comment:
It's better to make it somehow like this: `posts.filter(post => hashTags.every(tag => post.hashTags.includes(tag)))`